web/src/EnforcerPage.js

```diff
@@ -36,16 +36,18 @@ class EnforcerPage extends React.Component {
       let adapters = values[1];
       let models = values[2];
 
-      this.setState({
-        enforcer: enforcer,
-        adapters: adapters,
-        models: models,
-        adapter: adapters.filter(adapter => adapter.id === enforcer.adapter)[0],
-        model: models.filter(model => model.id === enforcer.model)[0],
-      });
-
-      this.getAdapterPolicies(enforcer.adapter);
-      this.getAdapterGroupingPolicies(enforcer.adapter);
+      if(enforcer != null) {

```

Comment:
we don't should lost `undefined`.

Reply:
@nodece for both `undefined` and `null` do we need not run this code?

Reply:
maybe we should add error message to the component. what do you think?

Reply:
@arteevraina

Reply:
both `undefined `and `null` will mean falsy value therefore we won't get the error. If we do something like this
```

if(enforcer) {
// do something
}
```

Please correct me if I am wrong.

Reply:
sure thing!